corehq/apps/app_manager/suite_xml/sections/sync_requests.py

```diff
@@ -0,0 +1,115 @@
+from corehq.apps.app_manager import id_strings
+from corehq.apps.app_manager.suite_xml.contributors import SuiteContributorByModule
+from corehq.apps.app_manager.suite_xml.sections.details import DetailsHelper
+from corehq.apps.app_manager.suite_xml.xml_models import (
+    Command,
+    Display,
+    PushFrame,
+    QueryData,
+    QueryPrompt,
+    SessionDatum,
+    Stack,
+    StackDatum,
+    SyncRequest,
+    SyncRequestPost,
+    SyncRequestQuery,
+    SyncRequestSession,
+    Text,
+)
+from corehq.apps.app_manager.xpath import XPath
+from corehq.apps.case_search.models import CALCULATED_DATA, MARK_AS_CLAIMED
+from corehq.util.view_utils import absolute_reverse
+
+
+class SyncRequestContributor(SuiteContributorByModule):
+    """
+    Adds a sync-request node, which sets the URL and query details for
+    synchronous searching and case claiming.
+
+    Search is available from the module's case list.
+
+    See "sync-request" in the `CommCare 2.0 Suite Definition`_ for details.
+
+
+    .. _CommCare 2.0 Suite Definition: https://github.com/dimagi/commcare/wiki/Suite20#sync-request
+
+    """
+    def get_module_contributions(self, module):
+        from corehq.apps.app_manager.models import AdvancedModule, Module
+
+        if isinstance(module, (Module, AdvancedModule)) and module.search_config:
+            domain = module.get_app().domain
+
+            details_helper = DetailsHelper(module.get_app())
+
+            sync_request = SyncRequest(
+                post=SyncRequestPost(
+                    url=absolute_reverse('claim_case', args=[domain]),
+                    data=[
+                        QueryData(
+                            key='case_id',
+                            ref="instance('results')/case[@case_type='{}'][@status='open'][0]/@case_id".format(
```

Comment:
Could be better to make `CaseSelectionXPath` able to use a different instance name